pina/geometry/simplex.py

```diff
@@ -0,0 +1,236 @@
+import torch
+from .location import Location
+from pina.geometry import CartesianDomain
+from pina import LabelTensor
+from ..utils import check_consistency
+
+class SimplexDomain(Location):
+    """PINA implementation of a Simplex."""
+
+    def __init__(self, simplex_matrix, sample_surface=False):
+        """
+        :param simplex_matrix: A matrix of LabelTensor objects representing
+            a vertex of the simplex (a tensor), and the coordinates of the
+            point (a list of labels).
+        :type simplex_matrix: list[LabelTensor]
+        :param sample_surface: A variable for choosing sample strategies. If
+            `sample_surface=True` only samples on the Simplex surface
+            frontier are taken. If `sample_surface=False`, no such criteria
+            is followed.
+        :type sample_surface: bool
+
+        .. warning::
+            Sampling for dimensions greater or equal to 10 could result
+            in a shrinking of the simplex, which degrades the quality
+            of the samples. For dimensions higher than 10, other algorithms
+            for sampling should be used.
+        
+        :Example:
+            >>> spatial_domain = SimplexDomain(
+                    [
+                        LabelTensor(torch.tensor([[0, 0]]), labels=["x", "y"]),
+                        LabelTensor(torch.tensor([[1, 1]]), labels=["x", "y"]),
+                        LabelTensor(torch.tensor([[0, 2]]), labels=["x", "y"]),
+                    ], sample_surface = True
+                )
+        """
+
+        # check consistency of sample_surface
+        check_consistency(sample_surface, bool)
+        self._sample_surface = sample_surface
+
+        # check consistency of simplex_dict
+        if not isinstance(simplex_matrix, list):
+            raise ValueError(f"{type(simplex_matrix).__name__} must be {list}.")
+        check_consistency(simplex_matrix, LabelTensor)
+        
+        # check consistency of labels
+        if simplex_matrix[0].labels != simplex_matrix[1].labels:
+            raise ValueError(f"Labels don't match.")
+
+        # vertices, vectors, coordinates
+        self._vertices_matrix = torch.cat(
+            [vertex.type(torch.FloatTensor) for vertex in simplex_matrix]
```

Comment:
I still don't understand why you are doing this. Can't you just do:

```
self._vertices_matrix = torch.cat(simplex_matrix).T
self._vertices_matrix.labels = simplex_matrix[0].labels
```

Does it work like that?

Reply:
If I do that, we run into the error `ValueError: the tensor has not the same number of columns of the passed labels.` If I try not adding labels, we run into the error `Expected a floating point or complex tensor as input. Got Long`, hence I either got to turn it into a `ComplexTensor`, `FloatTensor`, or `float`.

Reply:
Ok, the first error is a matter of stacking correctly, but you should be able to do it. Regarding the second one, where does specifically the error occur? Can you provide me a snippet of code as just example?

Reply:
I don't think we should be adding labels to the vertices matrix. And if we do want to, it will not have the same labels as the first vertex in `simplex_matrix`, which are the names of the variables in the coordinate system.

As for the second error, it is caused by the line `lambda_ = torch.linalg.solve(self._vectors_shifted, point_shift)`, when I use the line `self._vertices_matrix = torch.cat(simplex_matrix).T` as definition for `self._vertices_matrix`, as opposed to the line `self._vertices_matrix = torch.cat([vertex.type(torch.FloatTensor) for vertex in simplex_matrix]).T`, which throws no such error. Somehow, the `LabelTensor` object seems to be getting converted to a `Long` object, which is why I have to typecast it to a `torch.FloatTensor` object.